arm-servicebus/2015-08-01/swagger/servicebus.json

```diff
@@ -2642,6 +2889,196 @@
+		}
+      },
+      "description": "Represents a filter which is a composition of an expression and an action that is executed in the pub/sub pipeline."
+    },	
+	"CorrelationFilter": {
+      "type": "object",
+      "properties": {
+         "CorrelationId": {
+          "type": "string",
+          "description": "Gets the identifier of the correlation."
+        },
+		"MessageId": {
+          "type": "string",
+          "description": "Gets the identifier of the message."
+        },		
+		"To": {
+          "type": "string",
+          "description": "Gets the address to send to."
+        },
+		"ReplyTo": {
+		  "type": "string",
+          "description": "Gets the address of the queue to reply to."
+		},		
+		"Label": {
+		  "type": "string",
+          "description": "Gets the application specific label."
+		},
+		"SessionId": {
+		  "type": "string",
+          "description": "Gets the session identifier."
+		},
+		"ReplyToSessionId": {
+		  "type": "string",
+          "description": "Gets the session identifier to reply to."
+		},
+		"ContentType": {
+		  "type": "string",
+          "description": "Gets the content type of the message."
+		},		
+		"RequiresPreprocessing": {
+          "type": "boolean",
+          "description": "Gets a value indicating whether the CorrelationFilter expression requires preprocessing."
+        }
+      },
+      "description": "Represents the correlation filter expression."
+    },
+    "RuleAction": {
+      "type": "object",
+      "properties": {
+         "RequiresPreprocessing": {
+          "type": "boolean",
+          "description": "Gets a value that indicates whether the rule action requires preprocessing."
+        },
+         "Match": {
+          "type": "boolean",
+          "description": "Matches the BrokeredMessage against the FilterExpression."
+        }
+      },
+      "description": "Represents the filter actions which are allowed for the transformation of a message that have been matched by a filter expression."
+    },
+	"SqlRuleAction": {
+      "type": "object",
+      "properties": {
+         "SqlExpression": {
+          "type": "string",
+          "description": "Gets the SQL expression."
+        },
+		"CompatibilityLevel": {
+		"format": "int32",
+		"type": "integer",
+          "description": "This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20."
+		},
+		"RequiresPreprocessing": {
+          "type": "boolean",
+          "description": "Gets a value indicating whether the SQL filter expression requires preprocessing."
+        }
+		},
+      "description": "Represents set of actions written in SQL language-based syntax that is performed against a ServiceBus.Messaging.BrokeredMessage "
+      },
+	"CompositeAction": {
+      "type": "object",
+      "properties": {
+         "RequiresPreprocessing": {
```

Comment:
is this a required property or an optional property? If optional then what is the default value? If there is a default then it should be defined.